packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -105,11 +118,18 @@ class FlutterManifest {
   bool get isModule => moduleDescriptor != null;
 
   List<Map<String, dynamic>> get fontsDescriptor {
-   return _flutterDescriptor['fonts'] ?? const <Map<String, dynamic>>[];
+    final List<dynamic> fontList = _flutterDescriptor['fonts'];
+    return fontList == null
+        ? const <Map<String, dynamic>>[]
+        : fontList.map(castStringKeyedMap).toList();
```

Comment:
Type parameter on `map`...

This gets annoying. Perhaps we should have
```dart
Map<String, dynamic> toStringKeyedMap(dynamic json) {
  if (json == null) {
    return <String, dynamic>{};
  } else {
    final Map<dynamic, dynamic> map = json;
    return map.cast<String, dynamic>();
  }
}

List<Map<String, dynamic>> toStringKeyedMaps(dynamic json) {
  if (json == null) {
    return <Map<String, dynamic>>[];
  } else {
    final List<dynamic> list = json;
    return list.map<Map<String, dynamic>>(toStringKeyedMap).toList();
  }
}
```
Alternative naming: `toJsonObject`/`toJsonArray`.

Reply:
The empty map and list could be `const`, but then the semantics gets complicated: `toStringKeyedMap(null)` is immutable, whereas `toStringKeyedMap({})` is not. If we do use `const` we should return immutable objects in all cases.

Reply:
Let me look into creating a `toStringKeyedMaps()` method.

Reply:
@mravn-google, I may save that for a follow-up PR (`toStringKeyedMaps`).